Finally added remove in the project widget

henpe071 opened the pull request:
The items in the project widget are now removable. The function iterates recursively and deletes the item and all of its children.
Basic analyses now holds their IDs to make it simple to get them from the list of basic analyses.

Also, when adding a new video to the tree, if a folder is selected the video will now get added in that folder.

henpe071 committed:
Finally added remove on the items in the project widget

The items in the project widget are now removable. The function iterates recursively and deletes the item and all of its children.
Basic analyses now holds their IDs to make it simple to get them from the list of basic analyses.

Also, when adding a new video to the tree if a folder is selected, the video will now get added in that folder.

fredrikiselius commented on ViAn/GUI/projectwidget.cpp:
```diff
@@ -614,28 +625,61 @@ void ProjectWidget::remove_item() {
+            remove_tree_item(item->child(0));
+        }
+        // Remove the video from the list of videos
+        auto it = std::find(m_proj->get_videos().begin(), m_proj->get_videos().end(), v_proj);
```
Is the video project ever deleted? Else it needs to be. Same goes for all removed items, their underlying data structure needs to be properly deleted.

fredrikiselius reviewed (commented):
See comment below

henpe071 committed:
Added more deletes

Make sure the tree item and the data is deleted to avoid memory leaks.

henpe071 committed:
Cleaned both code and memory

Todo look at and maybe change analysis method with the proxy and the analysis for fixing the basic analysis destructor

henpe071 committed:
Added some more deletes

Added some more deletes to clean up all memory when closing a project

TODO clean overlay

henpe071 commented:
Fixed these things and more in the Remove-&-drawing-widget branch. It's mostly done except some things while deleting analyses, which I will solve after I get the code from your "remove temp map" branch.

fredrikiselius commented:
Shouldn't this be merged then? If so the branch should be removed

henpe071 commented:
Yeah this pull request is not really up for merging anymore. I'll close this one and put up the new one when its done (it needs drawing widget and the new temp folder fix). I mostly kept this up so you could see the comment and as a remainder for myself.
